src/api/factories/ListingItemFactory.ts

```diff
@@ -18,29 +19,45 @@ export class ListingItemFactory {
         this.log = new Logger(__filename);
     }
 
-    public async get(data: ItemMessageInterface, marketId: number): Promise<ListingItemCreateRequest> {
-        // const hash = crypto.SHA256(new Date().getTime().toString()).toString();
+    public async get(data: ItemMessageInterface): Promise<ListingItemCreateRequest> {
+        const escrowType = data.payment.escrow ? EscrowType[data.payment.escrow.type] : EscrowType.MAD;
+
```

Comment:
get(data: ItemMessageInterface): Promise<ListingItemCreateRequest>

get? where's getMessage and getModel?
passing in ItemMessage to get ListingItemCreateRequest is what you'd expect when calling from messageprocessor, but you were implementing the command side.